pennylane/ftqc/quantum_script_sequence.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains an implementation of a QuantumScriptSequence that can contain a series
+of tapes to be executed sequentially without resetting the devic state. For the
+FTQC/MBQC device prototype.
+"""
+import copy
+from functools import cached_property
+
+from pennylane.devices.preprocess import null_postprocessing
+from pennylane.ftqc import RotXZX
+from pennylane.measurements import Shots
+from pennylane.ops import RZ
+from pennylane.ops.functions import map_wires
+from pennylane.tape.qscript import QuantumScript
+from pennylane.transforms import transform
+
+
+@transform
+def split_at_non_clifford_gates(tape):
+    """The most basic implementation to ensure that we flush the buffer before
+    each non-Clifford gate. Pays no attention to wires/commutation, and splits
+    the tapes up more than necessary, but the logic is very simple."""
+    all_operations = [[]]
+
+    for op in tape.operations:
+        # if its a non-Clifford gate, and there are already ops in the list, add a new list
```

Comment:
Maybe I'm under-caffeinated, but can you describe why we have an empty list or lists, and we append more empty lists?

I guess I don't fully get what the inner/outer lists are representing.

Reply:
I added a bit more context in the docstring and rearranged the code a bit. I think it's clearer this way!